inst/include/plant/patch.h

```diff
@@ -295,6 +337,57 @@ ode::const_iterator Patch<T,E>::set_ode_state(ode::const_iterator it,
   return it;
 }
 
+// pre-cached environments, used for mutant runs
+template <typename T, typename E>
+ode::const_iterator Patch<T,E>::set_ode_state(ode::const_iterator it,
+                                              int index) {
+
+  it = ode::set_ode_state(species.begin(), species.end(), it);
+
+  environment = environment_cache[index];
+  it = environment.set_ode_state(it);
+
+  // std::cout << "mutant: etime " << environment.time << "; index " << index << std::endl;
+
+  compute_rates();
+  return it;
+}
+
+template <typename T, typename E>
+void Patch<T,E>::cache_ode_step() {
+  if(save_RK45_cache) { 
+    step_history.push_back(time());
+    environment_history.push_back(environment_cache);
+    // environment_cache.clear()
+  }
+}
+
+template <typename T, typename E>
+void Patch<T,E>::load_ode_step() {
```

Comment:
What's the difference between `load_ode_step` and `set_ode_state`?

Reply:
`load_ode_step` is called once per `ode_solver.h::step_to` and `set_ode_state` is called once for each Runga-Kutta sub-step in `ode_interface.h::derivs`.

Naming suggestions welcome.